peregrine/resources/submission/graphql/node.py

```diff
@@ -861,7 +861,8 @@ def resolve_datanode(self, info, **args):
         q = query_with_args(data_type, args, info)
         q_all.extend(q.all())
 
-    return [__gql_object_classes[n.label](**load_node(n, info)) for n in q_all]
+    limit = args['first']
```

Comment:
in the existing function `get_base_node_args()` there is a default value `first=graphene.Int(default_value=10)` but it probably shouldn't be hardcoded. I can change `get_base_node_args()` to use the global variable you mentioned

Reply:
Ooh thanks, I didn't remember this. Yeah un-hardcoding that would be a good idea I think. I was thinking as well, just in case `'first'` somehow does not get put in `args`, changing it to `args.get('first', some_default_value)` so that `args['first']` won't raise a KeyError.